api/api-rules/violation_exceptions.list

```diff
@@ -392,6 +392,7 @@ API rule violation: list_type_missing,k8s.io/kubelet/config/v1alpha1,CredentialP
 API rule violation: list_type_missing,k8s.io/kubelet/config/v1beta1,KubeletConfiguration,AllowedUnsafeSysctls
 API rule violation: list_type_missing,k8s.io/kubelet/config/v1beta1,KubeletConfiguration,ClusterDNS
 API rule violation: list_type_missing,k8s.io/kubelet/config/v1beta1,KubeletConfiguration,EnforceNodeAllocatable
+API rule violation: list_type_missing,k8s.io/kubelet/config/v1beta1,KubeletConfiguration,ReservedMemory
```

Comment:
Just for my own knowledge, what is this and why is it needed in this case?

Reply:
I remember I had some exceptions during the generation of the openapi spec, so I followed the documentation under https://github.com/kubernetes/kubernetes/tree/master/api/api-rules, as I can see, all parameters under the kubeletconfig with the array type currently added as exceptions. I did not dig too deep into it, but we can add it to our TODO list.

Reply:
After changing the structure of the type, this should no longer be necessary.

Reply:
IIUC the reserved memory is still a list, so we need this violation exception, I tried to remove it, but once I run `make generated_files UPDATE_API_KNOWN_VIOLATIONS=true` it was added again.

Reply:
Lists should not be a violation, because it should have a list type?

Reply:
I unsure what can I do now, when I tried to remove it and ran `make generated_files` I got
```
< API rule violation: list_type_missing,k8s.io/kubelet/config/v1beta1,KubeletConfiguration,ReservedMemory
ERROR: 
         KUBE API rule check failed. Reported violations differ from known violations. Please read api/api-rules/README.md to resolve the failure in api/api-rules/violation_exceptions.list.
```
Also, I can see that all variables of the type list included under the `violation_exceptions.list`.

@smarterclayton Do you have any idea how can I improve it?

Reply:
Oh.  This rule is being applied to the wrong objects.  ListType should only be required on top level kube objects (KubeletConfiguration), not sub types.  So you're fine to leave this exception as is, I'll track a separate bug on this to fix the validator.

Reply:
Thanks for checking!